### ISSUE TITLE
self host login, Access Denied: You do not have permission to sign in 

### ISSUE THREAD

**Issue by peter6055** [steven-tey/dub#225]
>@frankyw, you can remove all instances of `TWITTER_AUTH_TOKEN` since that's just used to render the static tweets on the landing page.
@GauravG8 as for hard coded `dub.sh` values, you'll set them to your custom domain (e.g. app.domain.com)
Closing this for now since it's a duplicate of #12
_Originally posted by @steven-tey in https://github.com/steven-tey/dub/issues/199#issuecomment-1566192848_
            

I could deploy dub on Vercel after following tutorial #199  above. Also, I added 'GOOGLE_CLIENT_ID' and 'GOOGLE_CLIENT_SECRET' as environmental variables on Vercel to allow Google login on my end.

However, I can successfully pass and grant sessions on Google OAuth. But, once the callback land on /api/auth/error?error=AccessDenied  and the error message "Access Denied: You do not have permission to sign in on" is displayed.

<img width="1474" alt="截圖 2023-05-30 上午2 08 08" src="https://user-images.githubusercontent.com/59694246/241771951-9b12021d-f089-4cc5-8138-4c31914e9c57.png">
<img width="1463" alt="截圖 2023-05-30 上午2 08 19" src="https://user-images.githubusercontent.com/59694246/241771929-7e671cb2-cb86-424f-97b2-1b18e459ae70.png">
<img width="1475" alt="截圖 2023-05-30 上午2 08 23" src="https://user-images.githubusercontent.com/59694246/241771946-1b006f8c-891c-45e6-bbd6-cfaaa041820d.png">
<img width="1487" alt="截圖 2023-05-30 上午2 06 13" src="https://user-images.githubusercontent.com/59694246/241771957-16b6bef8-217e-4a97-a659-7bd78e5a6671.png">

**Comment by creatorbegg** [comment]
I am getting the same error as you. Is there any news that has been updated since then?

@steven-tey Do you have any information on what the error is about this?

**Comment by my3t** [comment]
I am having the same issue with Google OAuth.

**Comment by Brams-s** [comment]
This might be due to the check for blacklisted emails. Temporarily commenting out this check in `pages/api/auth/[...nextauth].tsx` should do the trick:

```ts
callbacks: {
    signIn: async ({ user, account, profile }) => {
      // if (!user.email || (await isBlacklistedEmail(user.email))) {
      if(!user.email) {
        return false;
      }
      if (account?.provider === "google") {
        const userExists = await prisma.user.findUnique({
          where: { email: user.email },
          select: { name: true },
        });
        // if the user already exists via email,
        // update the user with their name and image from Google
        if (userExists && !userExists.name) {
          await prisma.user.update({
            where: { email: user.email },
            data: {
              name: profile?.name,
              // @ts-ignore - this is a bug in the types, `picture` is a valid on the `Profile` type
              image: profile?.picture,
            },
          });
        }
      }
      return true;
    },
    jwt: async ({ token, user, trigger, session }) => {
      // if (!token.email || (await isBlacklistedEmail(token.email))) {
      if(!token.email) {
        return {};
      }
      if (user) {
        token.user = user;
      }
      if (trigger === "update") {
        const refreshedUser = await prisma.user.findUnique({
          where: { id: token.sub },
        });
        token.user = refreshedUser;
        token.name = refreshedUser?.name;
        token.email = refreshedUser?.email;
        token.image = refreshedUser?.image;
      }
      return token;
    },
    session: async ({ session, token }) => {
      session.user = {
        // @ts-ignore
        id: token.sub,
        ...session.user,
      };
      return session;
    },
  },
```

**Comment by peter6055** [comment]
@Brams-s Your solution is correct. I am able to login to the dashboard right now. Thanks!

> This might be due to the check for blacklisted emails. Temporarily commenting out this check in `pages/api/auth/[...nextauth].tsx` should do the trick:
> 
> ```ts
> callbacks: {
>     signIn: async ({ user, account, profile }) => {
>       // if (!user.email || (await isBlacklistedEmail(user.email))) {
>       if(!user.email) {
>         return false;
>       }
>       if (account?.provider === "google") {
>         const userExists = await prisma.user.findUnique({
>           where: { email: user.email },
>           select: { name: true },
>         });
>         // if the user already exists via email,
>         // update the user with their name and image from Google
>         if (userExists && !userExists.name) {
>           await prisma.user.update({
>             where: { email: user.email },
>             data: {
>               name: profile?.name,
>               // @ts-ignore - this is a bug in the types, `picture` is a valid on the `Profile` type
>               image: profile?.picture,
>             },
>           });
>         }
>       }
>       return true;
>     },
>     jwt: async ({ token, user, trigger, session }) => {
>       // if (!token.email || (await isBlacklistedEmail(token.email))) {
>       if(!token.email) {
>         return {};
>       }
>       if (user) {
>         token.user = user;
>       }
>       if (trigger === "update") {
>         const refreshedUser = await prisma.user.findUnique({
>           where: { id: token.sub },
>         });
>         token.user = refreshedUser;
>         token.name = refreshedUser?.name;
>         token.email = refreshedUser?.email;
>         token.image = refreshedUser?.image;
>       }
>       return token;
>     },
>     session: async ({ session, token }) => {
>       session.user = {
>         // @ts-ignore
>         id: token.sub,
>         ...session.user,
>       };
>       return session;
>     },
>   },
> ```
